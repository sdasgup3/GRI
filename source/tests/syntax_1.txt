function displayAdjacencyMatrix(matrix, vertices)
{
  foreach(vertex; vertices.iterator()) {
    print( " " + vertex.__id + " ");
  }

  println("");

  it = vertices.iterator();

  foreach(line; matrix.iterator())
  {
    if(it.hasNext())
      vertex = it.next();

    print("" + vertex.__id + " |");
    foreach(item; line.iterator())
      print("" + item + " ");

    println("");
  }
}

function main(argv) {

  g = graph();
  g.loadFromFile("graphs/01_house.txt");

  matrix = g.getAdjacencyMatrix();
  vertices = g.getVertices();
  vertices1 = g.getVertices();
  displayAdjacencyMatrix(matrix, vertices);

  println("------ <-> -----------");
  foreach(start; vertices) {
    foreach(end; vertices) {
      edges = start<->end;
      foreach(edge; edges) {
        println("" + start.__id + " <-> " +  end.__id);
        println(edge.__w);
      }
    }
    println("");
  }
  println("------ -> -----------");

  foreach(start; vertices1) {
    foreach(end; vertices) {
      edges = start->end;
      foreach(edge; edges) {
        println("" + start.__id + " -> " +  end.__id);
        println(edge.__w);
      }
    }
    println("");
  }

  println("------ <-> all -----------");
  foreach(start; vertices) {
    edges = start<-> g.getVertices();
    println(" " + start.__id + " <-> All");
    foreach(edge; edges) {
      println(edge.__w);
    }
    println("");
  }
  println("------ -> all-----------");

  foreach(start; vertices1) {
    edges = start-> g.getVertices();
    println(" " + start.__id + " -> All");
    foreach(edge; edges) {
      println(edge.__w);
    }
    println("");
  }
  println("------  <- all-----------");

  foreach(start; vertices1) {
    edges = start <- g.getVertices();
    println("" + start.__id + " <- all" );
    foreach(edge; edges) {
      println(edge.__w);
    }
    println("");
  }
}

