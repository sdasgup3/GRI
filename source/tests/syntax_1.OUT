 0  1  2  3  4 
0 |0 1 0 0 1 
1 |0 0 1 1 0 
2 |1 0 0 0 0 
3 |1 0 1 0 0 
4 |0 1 0 0 0 
------ <-> -----------
0 <-> 1
0.1
0 <-> 2
2
0 <-> 3
3
0 <-> 4
0.4

1 <-> 0
0.1
1 <-> 2
1.2
1 <-> 3
1.3
1 <-> 4
4.1

2 <-> 0
2
2 <-> 1
1.2
2 <-> 3
3.2

3 <-> 0
3
3 <-> 1
1.3
3 <-> 2
3.2

4 <-> 0
0.4
4 <-> 1
4.1

------ -> -----------
0 -> 1
0.1
0 -> 4
0.4

1 -> 2
1.2
1 -> 3
1.3

2 -> 0
2

3 -> 0
3
3 -> 2
3.2

4 -> 1
4.1

------ <-> all -----------
 0 <-> All
3
0.1
2
0.4

 1 <-> All
0.1
1.3
4.1
1.2

 2 <-> All
3.2
2
1.2

 3 <-> All
3
1.3
3.2

 4 <-> All
0.4
4.1

------ -> all-----------
 0 -> All
0.1
0.4

 1 -> All
1.3
1.2

 2 -> All
2

 3 -> All
3
3.2

 4 -> All
4.1

------  <- all-----------
0 <- all
3
2

1 <- all
0.1
4.1

2 <- all
3.2
1.2

3 <- all
1.3

4 <- all
0.4

